devtoolsNG/xen/tools/gist.js

```diff
@@ -0,0 +1,37 @@
+const post = async (user, password, url, body) => {
+  const response = await fetch(url, {
+    method: 'POST',
+    //mode: 'cors',
+    cache: 'no-cache',
+    //credentials: 'include',
+    headers: {
+      'Authorization': `Basic ${btoa(`${user}:${password}`)}`,
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  });
+  console.log(response);
+  const json = await response.json();
+  console.log(json);
+  console.log(json?.html_url);
+  return json?.html_url;
+};
+
+// TODO(sjmiles): tokens should generally not available to the front-end, but this particular token
+// only allows GIST construction, so the risk is low. Also, this token is only for special use and
+// can be revoked at any time.
+const token = '9b603ddcb35d871ff93d393b381b3f49c9822a0f';
```

Comment:
When I made the gists earlier, I had to update the token and user to be for my GitHub account. Will that be the same here? If so, should there be some comments here that link to the GH docs on how to do that?

Reply:
This is not a long term thing, I'll probably just throw it away at some point.

Also, the token just went bad; the tool doesn't know who you are or what account you have. If I put a working token there it should work for anybody (as me).

Reply:
Turns out the reason the old token went bad is because if GitHub sees one of it's own tokens checked in, it is automatically revoked. I worked around this by splitting the token into two halves. I consider this a pious fraud because all this token can do is create GISTs and will be short-lived in any case.

After some reflection, perhaps I'm swimming upstream here, and I think I will follow your suggestion after all.